xknx/io/knxip_interface.py

```diff
@@ -91,10 +86,16 @@ def __init__(self, xknx, connection_config=ConnectionConfig()):
         if self.connection_config.connection_type == ConnectionType.AUTOMATIC:
             await self.start_automatic(
                 self.connection_config.scan_filter)
-        elif self.connection_config.connection_type == ConnectionType.ROUTING:
+        elif self.connection_config.connection_type == ConnectionType.ROUTING and \
+                self.connection_config.local_ip is not None:
             await self.start_routing(
                 self.connection_config.local_ip,
                 self.connection_config.bind_to_multicast_addr)
+        elif self.connection_config.connection_type == ConnectionType.ROUTING and \
+                self.connection_config.local_ip is None:
+            self.connection_config.scan_filter.routing = True
```

Comment:
I think this is a hack :) I think is kind of a `const` object and we should not modify it.

Reply:
Shall we instead instanciate a new `GatewayScanFilter()` ?

Reply:
🤔 now it seems even more hacky. 
Maybe we could define the GatewayScanFilter() when reading the configuration file / HA configuration. I'll look into it.